pkg/kubectl/cmd/create.go

```diff
@@ -57,16 +61,18 @@ func NewCmdCreate(f cmdutil.Factory, out io.Writer) *cobra.Command {
 				return
 			}
 			cmdutil.CheckErr(ValidateArgs(cmd, args))
-			cmdutil.CheckErr(cmdutil.ValidateOutputArgs(cmd))
-			cmdutil.CheckErr(RunCreate(f, cmd, out, options))
+			cmdutil.CheckErr(RunCreate(f, cmd, out, errOut, options))
 		},
 	}
 
 	usage := "to use to create the resource"
 	cmdutil.AddFilenameOptionFlags(cmd, options, usage)
 	cmd.MarkFlagRequired("filename")
 	cmdutil.AddValidateFlags(cmd)
-	cmdutil.AddOutputFlagsForMutation(cmd)
+	cmd.Flags().Bool("edit", false, "Edit the API resource before creating")
+	cmd.Flags().StringP("output", "o", "yaml", "Output format. If --edit is false, use -o name for shorter output (resource/name); Other usage will be ignored. If --edit is true, use one of: yaml|json. The default format is YAML.")
+	cmd.Flags().String("output-version", "", "Only relevant if --edit=true. Output the formatted object with the given group version (for ex: 'extensions/v1beta1').")
+	cmd.Flags().Bool("windows-line-endings", gruntime.GOOS == "windows", "Only relevant if --edit=true. Use Windows line-endings (default Unix line-endings)")
```

Comment:
> Perhaps we can print warnings to stderr when these three flags are specified without --edit?

It is not reasonable to check these flags for `--edit`, since `--dray-run` is using these flags, too.